Inline highlighting analytics

KarinaMendez2 opened the pull request.

Commit:
initial definitions to track
chosen color on creating
a note

TomWoodward committed:
fix type error

Commit:
initial changes to record
if default color is chosen
on create note

TomWoodward commented on src/app/content/highlights/components/EditCard.tsx:
```diff
@@ -61,7 +68,8 @@ const EditCard = React.forwardRef<HTMLElement, Props>((
 
-  const onColorChange = (color: HighlightColorEnum) => {
+  const onColorChange = (color: HighlightColorEnum, flag?: boolean) => {
+    services.analytics.createNote.track(book, flag ? 'default' : color);
```
its only a create down in the else on line 82, otherwise its an edit

TomWoodward replied:
if you look at some other events you'll see sometimes i put a selector in the event definition to make it easy to collect the data the event expects, this way if we decide we need more data we only have to change it in one place in the event definition. you can then access it here and pass it through. something like: 

```
import {useSelector} from 'react-redux';

...in the component somewhere
const noteEventData = useSelector(services.analytics.createNote.selector);
services.analytics.createNote.track(noteEventData, userSelected ? color : 'default');

```

TomWoodward commented on src/app/content/highlights/components/EditCard.tsx:
```diff
@@ -61,7 +68,8 @@ const EditCard = React.forwardRef<HTMLElement, Props>((
   React.useEffect(onClickOutside(element, isFocused, blurIfNotEditing), [isFocused, editingAnnotation]);
 
-  const onColorChange = (color: HighlightColorEnum) => {
+  const onColorChange = (color: HighlightColorEnum, flag?: boolean) => {
```
flag could be named something more useful like `userSelected` or `isDefault`

TomWoodward commented on src/helpers/analyticsEvents/highlighting/CreateNote.ts:
```diff
@@ -0,0 +1,21 @@
+  book: ReturnType<typeof selectContent['bookAndPage']>['book'],
+  color: string
+): AnalyticsEvent | void => {
+  const slug = book ? book.slug : 'unknown';
```
lets do the url path instead of just the book slug, check some other analytics events for examples

TomWoodward commented on src/app/content/highlights/components/ColorPicker.tsx:
```diff
@@ -62,25 +62,24 @@ const ColorButton = styled(({className, size, style, ...props}: ColorButtonProps
   flex-direction: row;
 `;
+
+export default ColorPicker;
```
i don't think any of the changes to this file are necessary

Commit:
added analytics labels and
regions for highlight edit
and create note, created
files to track events on react

Commit:
Merge remote-tracking branch 'origin/master' into highlights_analytics

TomWoodward commented on src/app/content/highlights/components/EditCard.tsx:
```diff
@@ -18,6 +22,7 @@ import onClickOutside from './utils/onClickOutside';
 
 interface Props {
   authenticated: boolean;
+  book: ReturnType<typeof selectContent['bookAndPage']>['book'];
```
i don't think you need this

TomWoodward commented on src/app/content/highlights/components/ColorPicker.tsx:
```diff
@@ -62,25 +62,24 @@ const ColorButton = styled(({className, size, style, ...props}: ColorButtonProps
 // tslint:disable-next-line:variable-name
-const ColorPicker = ({className, ...props}: Props) => {
-
+const ColorPicker = styled(({className, ...props}: Props) => {
```
were any of the changes in this file necessary?

KarinaMendez2 replied:
no.

TomWoodward commented on src/app/content/highlights/components/Confirmation.tsx:
```diff
@@ -52,6 +53,10 @@ const Confirmation = ({message, confirmMessage, confirmLink, always, onCancel, o
       ? {'data-analytics-region': props['data-analytics-region']}
       : {}
     }
+    {...props['data-analytics-label']
```
i don't think the label will do anything on this div

TomWoodward commented on src/helpers/analytics.ts:
```diff
@@ -13,6 +16,7 @@ const triggerEvent = <Args extends any[]>(event: (...args: Args) => (AnalyticsEv
   const analyticsEvent = event(...args);
 
   if (analyticsEvent) {
+    console.log(analyticsEvent.getGoogleAnalyticsPayload());
```
make sure to remove this when you're done testing

TomWoodward commented on src/helpers/analyticsEvents/highlighting/CreateNote.ts:
```diff
@@ -0,0 +1,23 @@
+import * as selectNavigation from '../../../app/navigation/selectors';
+import { AnalyticsEvent } from '../event';
+
+const createNote = 'REX highlighting - create note with chosen color';
```
its helpful for these to be short, so maybe just 
```suggestion
const createNote = 'REX highlighting - create note';
```

TomWoodward commented on src/helpers/analyticsEvents/highlighting/editColor.ts:
```diff
@@ -0,0 +1,23 @@
+import * as selectNavigation from '../../../app/navigation/selectors';
+import { AnalyticsEvent } from '../event';
+
+const editNote = 'REX highlighting - edit highlight color';
```
```suggestion
const editNote = 'REX highlighting - edit color';
```

TomWoodward commented on src/helpers/analyticsEvents/highlighting/editNote.ts:
```diff
@@ -0,0 +1,25 @@
+import * as selectNavigation from '../../../app/navigation/selectors';
+import { AnalyticsEvent } from '../event';
+
+const editNote = 'REX highlighting - edit highlight note';
```
```suggestion
const editNote = 'REX highlighting - edit note';
```

TomWoodward commented on src/helpers/analyticsEvents/highlighting/editNote.ts:
```diff
@@ -0,0 +1,25 @@
+import { AnalyticsEvent } from '../event';
+
+const editNote = 'REX highlighting - edit highlight note';
+const addNote = 'REX highlighting - note added';
```
to be consistent with the others
```suggestion
const addNote = 'REX highlighting - add note';
```

Commit:
Merge remote-tracking branch 'origin/master' into highlights_analytics

Commit:
reverted changes to ColorPicker, updated spec

Commit:
removed unused label, and
imports, reduced analytics
titles

Commit:
Merge remote-tracking branch 'origin/master' into highlights_analytics

TomWoodward committed:
Merge branch 'master' into highlights_analytics

TomWoodward commented on src/app/content/highlights/components/Card.tsx:
```diff
@@ -124,6 +124,7 @@ const Card = (props: Props) => {
   /> : <EditCard
     {...commonProps}
     authenticated={!!props.user}
+    book={props.book}
```
i don't think this is doing anything at this point

TomWoodward commented on src/app/content/highlights/components/ColorPicker.tsx:
```diff
@@ -51,7 +51,6 @@ const ColorButton = styled(({className, size, style, ...props}: ColorButtonProps
 )`
   cursor: pointer;
   margin: 0 ${cardPadding}rem ${cardPadding}rem 0;
-
```
i think stylelint will complain about not having a space here

TomWoodward committed:
rename file

TomWoodward committed:
Merge branch 'highlights_analytics' of github.com:openstax/books-web into highlights_analytics

TomWoodward committed:
cleanup

TomWoodward committed:
:camera:

TomWoodward committed:
Merge branch 'master' into highlights_analytics

TomWoodward committed:
better analytics regions

TomWoodward committed:
Merge branch 'highlights_analytics' of github.com:openstax/books-web into highlights_analytics
